test/Key/Derivation/SaltedS2kTest.php

```diff
@@ -27,8 +27,8 @@ public function setUp(): void
 
     public function testCalc()
     {
-        if (! extension_loaded('mhash')) {
-            $this->markTestSkipped('The mhash extension is not available');
+        if (! extension_loaded('hash')) {
```

Comment:
Though looking at the failed tests (`Use of undefined constant MHASH_ADLER32 - assumed 'MHASH_ADLER32') in my build and #8, I suspect there's a better solution.

Reply:
Perhaps we need to check for both `'hash' && PHP_VERSION_ID >= 80000` and `'mhash' && PHP_VERSION_ID < 80000`?

Reply:
The issue I'm having is that I can't reproduce this locally in either 7.3 or 7.4 using the official php docker images. This test isn't skipped for me, but I'm not seeing the same error either.

If I'm understanding the docs correctly, checking for the `mhash` extension is no longer possible as of PHP 7.0 (you should check for `hash` instead), and checking for `hash` is only necessary for <= PHP 7.3 because it's always enabled in later versions.

```
1) LaminasTest\Crypt\Key\Derivation\SaltedS2kTest::testCalc
Use of undefined constant MHASH_ADLER32 - assumed 'MHASH_ADLER32' (this will throw an Error in a future version of PHP)
```

It seems like this error is due strictly to the presence of the [`MHASH_ADLER32` constant](https://github.com/laminas/laminas-crypt/blob/f82bd2e61dd9a9aae81f2cdbaa078fa81c3fab8d/src/Key/Derivation/SaltedS2k.php#L22) because this test isn't even using that algo, and presumably the next constant in the list would trigger the same error.

It seems like [SaltedS2k::calc](https://github.com/laminas/laminas-crypt/blob/f82bd2e61dd9a9aae81f2cdbaa078fa81c3fab8d/src/Key/Derivation/SaltedS2k.php#L74) should be ported to the Hash extension instead, but there's no equivalent to `mhash_keygen_s2k` in that extension. I think we're in a weird spot where `mhash_keygen_s2k` is still available, but none of its constants are. Would it be acceptable to drop support for the salted S2K algo? Or should we attempt a userland solution?

Reply:
To add to this, I believe this issue has been a problem for awhile, but this test was always being skipped:

```
55) LaminasTest\Crypt\Key\Derivation\SaltedS2kTest::testCalc
The mhash extension is not available
```